src/software/ai/navigator/path_planner/hrvo/brute_force_nearest_neighbor_search.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <vector>
+
+/**
+ * Finds the neighbors of a candidate based on a comparator and radius
+ *
+ * @tparam T generic type for the candidate and its neighbors
+ * @tparam Func type for the comparator
```

Comment:
why do we have a Func type for the comparator?

In the conditional below, it looks like we're comparing it against radius^2 (which is a double)

Reply:
doesn't look like it's possible to parametrize std::function (easily)